crates/egui-winit/src/window_settings.rs

```diff
@@ -86,4 +83,58 @@ impl WindowSettings {
             *size = size.at_most(max_size);
         }
     }
+
+    pub fn clamp_window_to_sane_position<E>(
+        &mut self,
+        event_loop: &winit::event_loop::EventLoopWindowTarget<E>,
+    ) {
+        use std::ops::{Add, AddAssign, MulAssign, Sub};
+
+        if let (Some(position), Some(inner_size)) = (&mut self.position, &self.inner_size_points) {
+            let monitors = event_loop.available_monitors();
+            // default to primary monitor, in case the correct monitor was disconnected.
+            let mut monitor: winit::monitor::MonitorHandle = match event_loop.primary_monitor() {
+                Some(m) => m,
+                // No primary monitor, grab index 0 instead
+                _ => {
+                    match event_loop.available_monitors().nth(0) {
+                        Some(m) => m,
+                        None => return, // No monitors?
+                    }
+                }
+            };
```

Comment:
```suggestion
            let Some(mut monitor) = event_loop.primary_monitor().or_else(|| match event_loop.available_monitors().next()} else {
                return; // no monitors 🤷 
            };
```

Reply:
Oh nice! I didn't actually know about the or_else() function, thanks for the suggestion

Reply:
Had syntax issues with the suggested fix so I corrected it with what I thought you were going for, but it ended up a bit cluttered. 

`let mut active_monitor =
    if let Some(active_monitor) = event_loop.primary_monitor().or_else(|| event_loop.available_monitors().next()) { active_monitor }
    else {
        return; // no monitors 🤷
    };`